tests/test_core_model.py

```diff
@@ -281,39 +282,107 @@ def test_add_math_fails_marker_correct_position(self, simple_supply, eq_string):
 
 
 class TestOperateMode:
+    @contextmanager
+    def caplog_session(self, request):
+        """caplog for class/session-scoped fixtures.
+
+        See https://github.com/pytest-dev/pytest/discussions/11177
+        """
+        request.node.add_report_section = lambda *args: None
+        logging_plugin = request.config.pluginmanager.getplugin("logging-plugin")
+        for _ in logging_plugin.pytest_runtest_setup(request.node):
+            yield pytest.LogCaptureFixture(request.node, _ispytest=True)
+
     @pytest.fixture(scope="class")
     def plan_model(self):
+        """Solve in plan mode for the same overrides, to check against operate mode model."""
         model = build_model({}, "simple_supply,operate,var_costs,investment_costs")
         model.build(mode="plan")
         model.solve()
         return model
 
-    @pytest.fixture(scope="class")
-    def operate_model(self):
+    @pytest.fixture(
+        scope="class", params=[("6h", "12h"), ("12h", "12h"), ("16h", "20h")]
+    )
+    def operate_model_and_log(self, request):
```

Comment:
Just to understand what this is doing: this will create 3 versions of this fixture, meaning tests using it will be repeated 3 times?

Reply:
Exactly! It's a nice way to parameterise multiple tests using the same input params